### PR TITLE
[WIP] Add a `_flush_compilation_cache` for testing

### PR DESCRIPTION
**Pull request by Krovatkin**

### CODE REVIEW

File: torch/csrc/jit/runtime/profiling_graph_executor_impl.h [hunk 1]
```diff
@@ -14,6 +14,16 @@ struct ProfilingGraphExecutorImpl : public GraphExecutorImplBase {
   GraphExecutorState getDebugState() override;
   ~ProfilingGraphExecutorImpl() override = default;
 
+  void flushCompilationCache() {
```
Comment (eellison) [inline torch/csrc/jit/runtime/profiling_graph_executor_impl.h:17]: you might want to add some underscores here, or call it `debugFlushCompilationCache`. We dont want anyone else to use this really..

File: torch/csrc/jit/runtime/profiling_graph_executor_impl.h [hunk 2]
```diff
@@ -14,6 +14,16 @@ struct ProfilingGraphExecutorImpl : public GraphExecutorImplBase {
...
+    pr_.reset();
+    fallback_plan_.reset();
```
Comment (eellison) [inline torch/csrc/jit/runtime/profiling_graph_executor_impl.h:19]: I think some of these have mutexes that you're not accessing

**Review by eellison — APPROVED** [review]
Looks good as a debug mechanism, I think you should add more warnings or change the name of the function to indicate this is not for use outside of testing

File: torch/csrc/jit/runtime/profiling_graph_executor_impl.h (hunk 2 above)
Comment (Krovatkin) [reply torch/csrc/jit/runtime/profiling_graph_executor_impl.h:19]: Ah, yes! Thank you for pointing this out!

File: torch/csrc/jit/runtime/profiling_graph_executor_impl.h (hunk 1 above)
Comment (Krovatkin) [reply torch/csrc/jit/runtime/profiling_graph_executor_impl.h:17]: I'll rename it to `debug`

**Pull request closed without merging**